src/hip/hip_build_utils.cpp

```diff
@@ -79,6 +82,39 @@ char* generateFlagValue(MIOPEN_CMP_Flags_t x)
     return (char*)&(MIOPEN_CMP_FLAGS[x]);
 }
 
+// API to get MIOPEN AMD GCN Assembler Path Values.
+const char* getAMDGCNAssemblerPath()
+{
+    MIOPEN_CMP_Flags_t x          = MIOPEN_AMDGCN_Flag;
+    static const std::string path = generateCompilerPathValue(x);
+    return path.c_str();
+}
+
+// API to get MIOPEN OpenCL Compiler Path Values.
+const char* getOpenCLCompilerPath()
+{
+    MIOPEN_CMP_Flags_t x          = MIOPEN_OC_COMPILER_Flag;
+    static const std::string path = generateCompilerPathValue(x);
+    return path.c_str();
```

Comment:
So this would become
{
     static const char * path = generateCompilerPathValue("bin/clang-ocl");
     return path;
}
If you would prefer generateCompilerPathValue to return a std::string or a reference to one I would be happy with that as well.

Reply:
Thanks for the comments, updated the changes in latest version